custom_components/trakt_tv/apis/trakt.py

```diff
@@ -69,7 +69,43 @@ async def fetch_calendar(
             "get", f"calendars/{root}/{path}/{from_date}/{nb_days}"
         )
 
-    async def fetch_upcoming(self, trakt_kind: TraktKind, all_medias: bool):
+    async def fetch_watched(
+        self, exclude_shows: list
+    ):
+        response = await self.request(
+            "get", f"sync/watched/shows?extended=noseasons"
+        )
+        raw_shows = await response.json()
+        raw_medias = []
+        for show in raw_shows:
+            if show['show']['ids']['slug'] in exclude_shows:
+                continue
+            response_progress = await self.fetch_show_progress(show['show']['ids']['trakt'])
+            raw_episode = await response_progress.json()
+            if not (raw_episode.get('next_episode') is None):
+                if not (raw_episode['next_episode'].get('season') is None):
+                    response_episode = await self.fetch_show_informations(show['show']['ids']['trakt'],raw_episode['next_episode'].get('season'),raw_episode['next_episode'].get('number'))
```

Comment:
This can also fail if the key doesn't exist, we should ignore the show if the key doesn't exist.